### ISSUE TITLE
imagenet example

### ISSUE THREAD

**Issue by culurciello** [torchnet/torchnet#44]
hi guys, did anyone write an imagenet training script like this https://github.com/soumith/imagenet-multiGPU.torch
for torchnet?

**Comment by karandwivedi42** [comment]
https://github.com/karandwivedi42/imagenet-multiGPU.torchnet

**Comment by soumith** [comment]
@karandwivedi42 that's pretty nicely done. :)

**Comment by culurciello** [comment]
@karandwivedi42 great, thanks for sharing, that is pretty important piece and example!

**Comment by karandwivedi42** [comment]
@soumith @Atcold thanks for going over the code earlier. I have changed the code according to your suggestions. Works just like imagenet-multiGPU.torch now.

**Comment by Atcold** [comment]
@karandwivedi42, that's great! Do you achieve similar speed?

**Comment by karandwivedi42** [comment]
@Atcold I don't know how to.

**Comment by karandwivedi42** [comment]
@Atcold: on 4GPUs
* around 4000 seconds per train epoch
* 33.76% - Top 1 % ACCURACY after 7 epochs

How does it look?

**Comment by Atcold** [comment]
I think @culurciello is looking into that right now.
He said our image loading takes 10s of second whereas the one you have here is 10 times slower. Or at least this is how it seems to be.

**Comment by karandwivedi42** [comment]
@Atcold You might have to use this: https://gist.github.com/szagoruyko/4cbbba5ac8b53b0fe32f43e7b3d0cda6

**Comment by Atcold** [comment]
We just profiled it. It takes `1370`s per epoch (`10,000` batches).

**Comment by karandwivedi42** [comment]
How does that compare to imagenet-multiGPU.torch?

**Comment by Atcold** [comment]
@karandwivedi42 , I've just updated my post.
